src/sentry/sentry_metrics/consumers/indexer/common.py

```diff
@@ -154,5 +155,8 @@ def join(self, timeout: float | None = None) -> None:
                 last.committable,
             )
 
+        if options.get("sentry-metrics.indexer.drop-work-in-multiprocessing"):
+            timeout = 0.0
+
```

Comment:
Does surrounding this with an option flag does anything? I thought this code only get called once during startup. So if we change this option value it would still take a redploy for it to go into effect, right?

Reply:
Yep that's right, a redeploy or rebalance. I guess my thought process was to just make this functionality restricted to one environment in the beginning. But you're right that a switch of the option doesn't have an immediate impact. I'll remove it once I double-check that things are okay in the initial test.